turbopack/crates/turbo-tasks/src/manager.rs

```diff
@@ -1704,11 +1703,9 @@ pub fn current_task_for_testing() -> TaskId {
 }
 
 /// Marks the current task as dirty when restored from persistent cache.
-pub fn mark_dirty_when_persisted() {
+pub fn mark_session_dependent() {
     with_turbo_tasks(|tt| {
-        tt.mark_own_task_as_dirty_when_persisted(current_task(
-            "turbo_tasks::mark_dirty_when_persisted()",
-        ))
+        tt.mark_own_task_as_session_dependent(current_task("turbo_tasks::mark_session_dependent()"))
     });
 }
```

Comment:
Future work: While this imperative API is the most powerful option, it may also make sense put this as an option on the `#[turbo_tasks::function]` macro. I think that'd be a more discoverable place to put this.